File: web/src/lib/schema/club.ts
import { z } from 'zod'

const ClubSchema = z.object({
	Id: z.number(),
	Name: z.string().min(1).max(50),
	AddressLine1: z.string().max(50),
	PostalTown: z.string().max(50),
	PostCode: z.string().max(10)
})

type Club = z.infer<typeof ClubSchema>
export { ClubSchema, Club }

File: Clubz/Models/Club.cs
namespace Clubz.Models;

using System.ComponentModel.DataAnnotations;

public class Club
{
  public int Id { get; set; }
  [Required]
  public string Name { get; set; }
  [Required]
  public string AddressLine1 { get; set; }
  [Required]
  public string PostalTown { get; set; }
  [Required]
  public string Postcode { get; set; }
}  